builtin/credential/userpass/stepwise_test.go

```diff
@@ -0,0 +1,81 @@
+package userpass
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/policyutil"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/hashicorp/vault/sdk/testing/stepwise"
+	dockerEnvironment "github.com/hashicorp/vault/sdk/testing/stepwise/environments/docker"
+)
+
+func TestAccBackend_stepwise_UserCrud(t *testing.T) {
+	customPluginName := "my-userpass"
+	envOptions := &stepwise.MountOptions{
+		RegistryName:    customPluginName,
+		PluginType:      stepwise.PluginTypeCredential,
+		PluginName:      "userpass",
+		MountPathPrefix: customPluginName,
+	}
+	stepwise.Run(t, stepwise.Case{
+		Environment: dockerEnvironment.NewEnvironment(customPluginName, envOptions),
+		Steps: []stepwise.Step{
+			testAccStepwiseUser(t, "web", "password", "foo"),
+			testAccStepwiseReadUser(t, "web", "foo"),
+			testAccStepwiseDeleteUser(t, "web"),
+			testAccStepwiseReadUser(t, "web", ""),
+		},
+	})
+}
+
+func testAccStepwiseUser(
+	t *testing.T, name string, password string, policies string) stepwise.Step {
```

Comment:
Nit: I wonder if most of these strings should be put into a struct?  It's repeated a bunch in other tests and would make the function more compact or turn it into a method.

Reply:
🤔 I'm going to defer this for now, I'm not seeing this method used often, just this test. Am I mistaken?

Reply:
I was referring to `name string, password string, policies string` being used in most of the tests below.  No worries!